docs/src/examples.md

```diff
@@ -86,6 +86,45 @@ julia> round.(vcov(ols); digits=5)
   0.38889  -0.16667
  -0.16667   0.08333
 ```
+By default, the `lm` method uses the `Cholesky` factorization which is known as fast but numerically unstable, especially for ill-conditioned design matrices. You can use the `method` keyword argument to apply a more stable `QR` factorization method.
```

Comment:
everywhere in docs - should code `Cholesky` be `:cholesky`. The same for `QR` and `:qr`. In sentences where Cholesky an QR are used in mathematical sense I think they do not need to be backquoted.

Reply:
also - is this sentence still true? We discussed that automatic switching to `:qr` would be performed if `:cholesky` failed.

Reply:
@bkamins if we're tying this into [the 2.0 release](https://github.com/JuliaStats/GLM.jl/issues/500) (which it looks that way), then we can just default to QR.

Reply:
As you prefer. I think that most users would prefer to have slower but reliable option as a default.

Reply:
Cholesky fails for numerical reasons often (and this is unrelated to statistical error):
```
julia> x = [1, 2, 3]
3-element Vector{Int64}:
 1
 2
 3

julia> y = [1, 2, 4]
3-element Vector{Int64}:
 1
 2
 4

julia> lm(@formula(y~x), (; x, y))
StatsModels.TableRegressionModel{LinearModel{GLM.LmResp{Vector{Float64}}, GLM.DensePredChol{Float64, LinearAlgebra.CholeskyPivoted{Float64, Matrix{Float64}, Vector{Int64}}}}, Matrix{Float64}}

y ~ 1 + x

Coefficients:
─────────────────────────────────────────────────────────────────────────
                 Coef.  Std. Error      t  Pr(>|t|)  Lower 95%  Upper 95%
─────────────────────────────────────────────────────────────────────────
(Intercept)  -0.666667    0.62361   -1.07    0.4788   -8.59038    7.25704
x             1.5         0.288675   5.20    0.1210   -2.16797    5.16797
─────────────────────────────────────────────────────────────────────────

julia> lm(@formula(y~x), (x=10^9*x, y=y))
StatsModels.TableRegressionModel{LinearModel{GLM.LmResp{Vector{Float64}}, GLM.DensePredChol{Float64, LinearAlgebra.CholeskyPivoted{Float64, Matrix{Float64}, Vector{Int64}}}}, Matrix{Float64}}

y ~ 1 + x

Coefficients:
─────────────────────────────────────────────────────────────────────────────────────
                  Coef.     Std. Error       t  Pr(>|t|)      Lower 95%     Upper 95%
─────────────────────────────────────────────────────────────────────────────────────
(Intercept)  0.0         NaN            NaN       NaN     NaN            NaN
x            1.21429e-9    1.12938e-10   10.75    0.0085    7.28351e-10    1.70022e-9
─────────────────────────────────────────────────────────────────────────────────────
```

Reply:
If the scaling of the variables is extreme then indeed the rank estimate of the pivoted cholesky would suggest a reduced rank. I don't think that is the typical reason for rank deficiency, though. I think that is multicollinearity. Notice also that the Cholesky doesn't actually fail in your example so it's a consequence of the chosen tolerance
```
julia> X4 = [ones(3) 1e100*x]
3×2 Matrix{Float64}:
 1.0  1.0e100
 1.0  2.0e100
 1.0  3.0e100

julia> cholesky(X4'X4)
Cholesky{Float64, Matrix{Float64}}
U factor:
2×2 UpperTriangular{Float64, Matrix{Float64}}:
 1.73205  3.4641e100
  ⋅       1.41421e100
```

Reply:
> I don't think that is the typical reason for rank deficiency, though. I think that is multicollinearity. Notice also that the Cholesky doesn't actually fail in your example so it's a consequence of the chosen tolerance

I am in the middle of porting all examples from Wooldridge 7e to GLM.jl. In first 9 chapters I GLM.jl fails to correctly estimate the model I think 3 or 4 times. In each case the reason is the case I have presented (but in all cases on real data - not artificial like my example above).

I was taking perspective of a user (I have never maintained GLM.jl) - and I did not investigate what caused to the problem, but I knew that manually doing QR worked. If the problem is with tolerance then I think we should also fix it.

Reply:
Its possible that my gut feeling isn't right and that people use highly variably scaled variables more often that I have thought. It might also be that we shouldn't reduce the rank unless strictly required. Do you have the failing examples from Wooldridge handy? Then I can take a look. I have the 6th edition so I'd think the examples are mostly identical.

Reply:
I just realized that you'd already mentioned the example in the main thread. Sorry about that.

I just tried the first one and indeed adjusting the tolerance fixes the issue.

Current:
```julia
julia> lm(@formula(rdintens ~ sales + salessq), rdchem_df)
LinearModel

rdintens ~ 1 + sales + salessq

Coefficients:
───────────────────────────────────────────────────────────────────────────────────────
                    Coef.     Std. Error       t  Pr(>|t|)      Lower 95%     Upper 95%
───────────────────────────────────────────────────────────────────────────────────────
(Intercept)   0.0          NaN            NaN       NaN     NaN            NaN
sales         0.000852509    0.000156784    5.44    <1e-05    0.000532313    0.00117271
salessq      -1.97385e-8     4.56287e-9    -4.33    0.0002   -2.90571e-8    -1.04199e-8
───────────────────────────────────────────────────────────────────────────────────────
```
and with `tol=0` in https://github.com/JuliaStats/GLM.jl/blob/57669ef89de3588ffdbf757d82fd28d5c71f5177/src/linpred.jl#L107
```julia
julia> lm(@formula(rdintens ~ sales + salessq), rdchem_df)
LinearModel

rdintens ~ 1 + sales + salessq

Coefficients:
─────────────────────────────────────────────────────────────────────────────────
                    Coef.   Std. Error      t  Pr(>|t|)    Lower 95%    Upper 95%
─────────────────────────────────────────────────────────────────────────────────
(Intercept)   2.61251      0.429442      6.08    <1e-05   1.73421     3.49082
sales         0.000300571  0.000139295   2.16    0.0394   1.56805e-5  0.000585462
salessq      -6.94594e-9   3.72614e-9   -1.86    0.0725  -1.45667e-8  6.7487e-10
─────────────────────────────────────────────────────────────────────────────────
```
which matches the result from R. I'll file a separate issue about that but I don't think we should change any default or introduce `:auto` in this PR.

Reply:
OK - as I have said - I have not tested it in detail (I am reporting from user's perspective 😄).
@mousum-github - can you then please give examples when Cholesky fails in practice and QR does not fail?

@andreasnoack - regarding tolerance: yes, in particular it would be worth to dicsuss when setting such tolerance would lead to the reverse problems (i.e. not detectin multicollinearity when it is present; these issues are unfortunately tricky).

Reply:
I've opened https://github.com/JuliaStats/GLM.jl/pull/518 to see the effect of changing the tolerance and we can't discuss the details there but my current thinking is that we should probably only drop variables when it's numerically necessary. It's indeed tricky, though.